src/editors/containers/VideoEditor/components/VideoSettingsModal/components/DurationWidget/hooks.test.js

```diff
@@ -0,0 +1,328 @@
+      it('updates unsavedStartStopTimes startTime values to 1 second before stopTime', () => {
+        hook({
+          ...props,
+          index: testValidIndex,
+          inputString: '00:10:00',
+        });
+        expect(props.setUnsavedStartStopTimes).toHaveBeenCalledWith({
+          ...props.unsavedStartStopTimes,
+          [testValidIndex]: '00:09:59',
+        });
+        expect(props.setReduxStartStopTimes).toHaveBeenCalledWith({
+          ...props.reduxStartStopTimes,
+          [testValidIndex]: 599000,
+        });
+      });
+    });
+    describe('if the passed stopTime is before (or equal to) the stored startTime', () => {
+      it('updates unsavedStartStopTimes stopTime values to 1 second after startTime', () => {
+        hook({
+          ...props,
+          index: testStopIndex,
+          inputString: '00:00:22',
+        });
+        expect(props.setUnsavedStartStopTimes).toHaveBeenCalledWith({
+          ...props.unsavedStartStopTimes,
+          [testStopIndex]: '00:00:24',
+        });
+        expect(props.setReduxStartStopTimes).toHaveBeenCalledWith({
+          ...props.reduxStartStopTimes,
+          [testStopIndex]: 24000,
+        });
+      });
+    });
+  });
+  describe('onDurationChange', () => {
+    beforeEach(() => {
+      props = {
+        duration: { startTime: '00:00:00' },
+        index: 'startTime',
+        val: 'vAl',
+      };
+      hook = hooks.onDurationChange;
+    });
+    it('returns duration with no change if duration[index] does not match HH:MM:SS format', () => {
+      const badChecks = [
+        'ab:cd:ef', // non-digit characters
+        '12:34:567', // characters past max length
+      ];
+      badChecks.forEach(val => expect(hook(props.duration, props.index, val)).toEqual(props.duration));
+    });
+    it('returns duration with an added \':\' after 2 characters when caret is at end', () => {
+      props.duration = { startTime: '0' };
+      props.val = '00';
+      document.activeElement.selectionStart = props.duration[props.index].length + 1;
+      expect(hook(props.duration, props.index, props.val)).toEqual({ startTime: '00:' });
+    });
+    it('returns duration with an added \':\' after 5 characters when caret is at end', () => {
+      props.duration = { startTime: '00:0' };
+      props.val = '00:00';
+      document.activeElement.selectionStart = props.duration[props.index].length + 1;
+      expect(hook(props.duration, props.index, props.val)).toEqual({ startTime: '00:00:' });
+    });
+  });
+  describe('onDurationKeyDown', () => {
+    beforeEach(() => {
+      props = {
+        duration: { startTime: '00:00:00' },
+        index: 'startTime',
+        event: 'eVeNt',
+      };
+      hook = hooks.onDurationKeyDown;
+    });
+    it('enter event: calls blur()', () => {
+      props.event = { key: 'Enter' };
+      const blurSpy = jest.spyOn(document.activeElement, 'blur');
+      hook(props.duration, props.index, props.event);
```

Comment:
Not sure about the naming here - from the name, it is pretty unclear what a method `hook(...)` is supposed to do. Create a new react hook?